### PR TITLE
Tidy support for 8/16 mics and parallel decimators

### PR DESCRIPTION
**Pull request by ed-xmos**
https://xmosjira.atlassian.net/browse/AP-12

### CODE REVIEW

**Comment by ed-xmos** [comment]
This adds the 16ch de-interleave which will close this https://github.com/xmos/lib_mic_array/pull/193. It has a passing unit test and I have tested it in hardware too.
There is also a sample app to show use of the parallel decimators which does not run without specific hardware and so it is not so useful. Perhaps in time, the 16 mic aggregator could replace it or perhaps the 16 mic sum/delay example. However I do not think it is urgent.

**Comment by ed-xmos** [comment]
This will also allow deletion of https://github.com/xmos/lib_mic_array/tree/jcarrier/feature/16-mic-support

File: demos/demo_par_decimator/src/main.xc
Comment (mbanth) [inline demos/demo_par_decimator/src/main.xc:21]: Obviously, it's convenient to place the entire main function in an unsafe scope.  However, for someone with less than an expert's understanding of XC, I believe it will be more helpful to only use unsafe scope where it's actually necessary.  In this function, that appears to be line 61 and no where else.

File: doc/programming_guide/src/vanilla_api.rst
Comment (mbanth) [inline doc/programming_guide/src/vanilla_api.rst:165]: Good catch of the previous mis-spelling.  However, I think you mean `PORT_MCLK_IN_OUT`.

File: lib_mic_array/src/deinterleave16.S
Comment (mbanth) [inline lib_mic_array/src/deinterleave16.S:1]: Copyright date?

File: tests/unit/src/test_deinterleave16.c
Comment (mbanth) [inline tests/unit/src/test_deinterleave16.c:1]: Copyright date range?

File: tests/unit/src/test_deinterleave16.c
Comment (mbanth) [inline tests/unit/src/test_deinterleave16.c:43]: Please either remove these commented out lines or add a comment explaining why they have been left in.

File: demos/demo_par_decimator/XCORE-AI-EXPLORER.xn
Comment (mbanth) [inline demos/demo_par_decimator/XCORE-AI-EXPLORER.xn:112]: I'm a bit surprised at the inclusion of the Type, PageSize, SectorSize and NumPages arguments.  I thought that recent changes Tools 15.2 meant that we no longer needed these arguments and that their inclusion caused problems.  However, my notoriously bad memory means that I am not sure of the facts, and I see that the other .xn files in this repository look similar to this one as far as this line goes.

**Review by mbanth — CHANGES_REQUESTED** [review]

File: demos/common/src/dac_port.c
Comment (ed-xmos) [reply demos/common/src/dac_port.c:10]: Yes since it is accessed by member functions aic3204_board_init and aic3204_reg_write. It's written in the RTOS style.

File: demos/common/src/dac_port.c
Comment (ed-xmos) [reply demos/common/src/dac_port.c:24]: fixed

File: demos/demo_par_decimator/src/app_i2s.c
Comment (ed-xmos) [reply demos/demo_par_decimator/src/app_i2s.c:145]: It looks like John thought he might control this externally eventually but never did. I'll tidy it.

File: demos/demo_par_decimator/src/main.xc
Comment (ed-xmos) [reply demos/demo_par_decimator/src/main.xc:99]: I know - it's lib_xua levels of #ifs and not pretty.. I have added some comments but really didn't want to refactor all of it at this point given current priorities and the fact It's for custom HW anyhow.

File: tests/unit/src/test_deinterleave16.c
Comment (ed-xmos) [reply tests/unit/src/test_deinterleave16.c:1]: good catch

File: tests/unit/src/test_deinterleave16.c
Comment (ed-xmos) [reply tests/unit/src/test_deinterleave16.c:43]: They are useful so added a comment

File: demos/demo_par_decimator/XCORE-AI-EXPLORER.xn
Comment (ed-xmos) [reply demos/demo_par_decimator/XCORE-AI-EXPLORER.xn:112]: You are correct - this shows it https://github.com/xmos/sw_xvf3800/commit/4349ede0fd3b0527462de01da0402cdcc5322aa9

I will fix!

**Review by xhuw — APPROVED** [review]
Your changes since Michael's review appear to solve all of his comments. I didn't look at any previous changes.
You might need to get a super admin to work around Michael's "needs work" this approval might do the job though

**Comment by xhuw** [comment]
I "dismissed" Michaels review so now it's mergeable

**Pull request merged**
